apm-agent-core/src/main/java/co/elastic/apm/agent/metrics/builtin/SystemMetrics.java

```diff
@@ -116,12 +131,42 @@ public double get() {
             }
         });
 
-        metricRegistry.addUnlessNan("system.memory.actual.free", Collections.<String, String>emptyMap(), new DoubleSupplier() {
-            @Override
-            public double get() {
-                return invoke(freeMemory);
-            }
-        });
+        if (meminfo.canRead()) {
+            metricRegistry.addUnlessNan("system.memory.actual.free", Collections.<String, String>emptyMap(), new DoubleSupplier() {
+                final List<WildcardMatcher> relevantLines = Arrays.asList(
+                    caseSensitiveMatcher("MemAvailable:*"),
+                    caseSensitiveMatcher("MemFree:*"),
+                    caseSensitiveMatcher("Buffers:*"),
+                    caseSensitiveMatcher("Cached:*"));
+
+                @Override
+                public double get() {
+                    Map<String, Long> memInfo = new HashMap<>();
+                    try (BufferedReader fileReader = new BufferedReader(new FileReader(meminfo))) {
+                        for (String memInfoLine = fileReader.readLine(); memInfoLine != null && !memInfoLine.isEmpty(); memInfoLine = fileReader.readLine()) {
+                            if (WildcardMatcher.isAnyMatch(relevantLines, memInfoLine)) {
+                                final String[] memInfoSplit = StringUtils.split(memInfoLine, ' ');
+                                memInfo.put(memInfoSplit[0], Long.parseLong(memInfoSplit[1]));
```

Comment:
`* 1024`?
maybe also check it ends with "kB", and ignore otherwise?

Reply:
🤦‍♂️